File: DorotOlam/Models/Legacy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DorotOlam.Models
{
    public class Legacy
    {
        //public string Affiliation { get; set; }
        //public string Template { get; set; }

        public List<StudyPlace> studyPlace { get; set; } = new List<StudyPlace>();
        public List<Teacher> rabbis { get; set; } = new List<Teacher>();

        public List<Student> student { get; set; } = new List<Student>();

        public List<Work> work { get; set; } = new List<Work>();

        public List<MoreLink> moreLink { get; set; } = new List<MoreLink>();

    }
       
        public class Teacher
        {
            public string Name { get; set; }
            public string Address { get; set; }
            //public string Role { get; set; }
        }
        public class Student
        {
            public string Name { get; set; }
            public string Address { get; set; }
           // public string Role { get; set; }
        }

        public class Work
        {
        public string Name { get; set; }
        public string Address { get; set; }
         public string Role { get; set; }
    }
    public class MoreLink
    {
        public string Description { get; set; }
        public string Link { get; set; }
    }
    public class StudyPlace
    {
        public string Name { get; set; }
        public string Address { get; set; }
        public string Role { get; set; }
    }

}

File: DorotOlam/_090DorotOlam/src/app/memorial/models/legacy/legacy.ts
import { MoreLink } from '../moreLink/more-link';
import { Occupation } from '../occupation/occupation';
import { Student } from '../student/student';
import { StudyPlace } from '../studyPlace/studyplace';
import { Teacher } from '../teacher/teacher';
import { Work } from '../work/work';

export class Legacy {
    affiliation: string="";
    template: string="";
    studyPlace: StudyPlace[]=[];
    occupation: Occupation[]=[];
    teachers: Teacher[]=[];
    students: Student[]=[];
    works: Work[]=[];
    moreLink: MoreLink[]=[];
}
